src/commands/keys.rs

```diff
@@ -0,0 +1,145 @@
+pub struct Register {
+    /// The location of the public key to add
+    #[structopt(long = "public")]
+    public_key_location: String,
+
+    /// The location of the private key to add
+    #[structopt(long = "private")]
+    private_key_location: String,
+}
+
+/// Deletes a key to wapm
+#[derive(StructOpt, Debug)]
+pub struct Delete {
+    /// The identifier of the public key
+    public_key: String,
+}
+
+pub fn keys(options: KeyOpt) -> Result<(), failure::Error> {
+    let mut key_db = open_keys_db()?;
+    match options {
+        KeyOpt::List(List { all }) => {
+            // query server?
+            let keys = get_personal_keys_from_database(&key_db)?;
+            if all {
+                let wapm_public_keys = get_wapm_public_keys_from_database(&key_db)?;
+                match (keys.is_empty(), wapm_public_keys.is_empty()) {
+                    (true, true) => println!("No keys found"),
+                    (true, false) => {
+                        println!("{}", create_wapm_public_key_table(wapm_public_keys)?);
+                    }
+                    (false, true) => {
+                        println!("{}", create_personal_key_table(keys)?);
+                    }
+                    (false, false) => {
+                        println!("PERSONAL KEYS:\n{}", create_personal_key_table(keys)?);
+                        println!(
+                            "\nWAPM PUBLIC KEYS:\n{}",
+                            create_wapm_public_key_table(wapm_public_keys)?
+                        );
+                    }
+                }
+            } else {
+                if keys.is_empty() {
+                    println!("No personal keys found");
+                } else {
+                    println!("{}", create_personal_key_table(keys)?);
+                }
+            }
+        }
+        KeyOpt::Register(Register {
+            public_key_location,
+            private_key_location,
+        }) => {
+            // mutate server
+            add_personal_key_pair_to_database(
+                &mut key_db,
+                public_key_location.clone(),
+                private_key_location.clone(),
+            )?;
+            println!("Key pair successfully added!")
+        }
+        KeyOpt::Delete(Delete { public_key }) => {
+            let full_public_key =
+                get_full_personal_public_key_by_pattern(&key_db, public_key.clone())?;
+            warn!(
+                "You are about to delete the key pair associated with {:?} from wapm. This cannot be undone.",
+                &full_public_key
+            );
+            print!("Please confirm that you want to permanently delete this key pair from wapm:\n[y/n] ");
+            std::io::stdout().flush()?;
+            let mut input_str = String::new();
+            std::io::stdin().read_line(&mut input_str)?;
+            if let Some(first_char) = input_str.to_lowercase().chars().next() {
+                if first_char == 'y' {
```

Comment:
I think rather than getting only the first one, we can just accept `'y'` or `'yes'` (in case someone puts other thing that starts with y, like yandex :P). Small nit though, we can merge without this 👍